### PR TITLE
Voa fix date

### PR DESCRIPTION
**Pull request by awong-dev**

### CODE REVIEW

**Comment by awong-dev** [comment]
@akainic please only review comit a97e240.  The other two commits from Karma are in another PR (#1202)

@joesoftware for awareness.

**Comment by awong-dev** [comment]
Rebased and all ready for review!

File: _health-care/_js/_form.js [hunk 1]
```diff
@@ -6,29 +6,30 @@ function identityTransform(formData, arg) { return arg; }
...
+  return extractTransform(formData, arg + "[year]");
 }
 function extractDateWithSlashTransform(formData, arg) {
-  let date = new Date(extractTransform(formData, arg));
   let dateString = "";
 
-  if (date.getMonth() < 10) {
+  let month = extractMonthTransform(formData, arg);
+  let date = extractMonthTransform(formData, arg);
+  if (Number(month) < 10) {
```
Comment (akainic) [inline _health-care/_js/_form.js]: Should we somehow check if `month` and `date` are blank before modifying `dateString`? Not sure what's better to submit if the field is blank: `''` or `'0/0/'` (`year` isn't being set to 0).

**Comment by akainic** [comment]
@awong-dev Just reviewed, everything LGTM except for one comment/question!

File: _health-care/_js/_form.js (hunk 1 above)
Comment (awong-dev) [inline _health-care/_js/_form.js]: Good catch. Sounds like time for a unittest!

BTW, for a error like this, I would prefer you with holding LGTM. It's not polished enough for merge IMO. :)

**Comment by awong-dev** [comment]
Back atcha @akainic.  Added some test framework changes which might be helpful with your PR #1209 too.

File: spec/javascripts/health-care/health-care-unittest.spec.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+describe('Health Care Form UnitTests', function() {
+  before(function(){
+    fixture.setBase('spec/fixtures')
+  });
+  afterEach(function(){
+    fixture.cleanup();
+  });
+
+  describe('Extract Functions', function() {
+    it('extracts date form a composite date component', function() {
```
Comment (akainic) [inline spec/javascripts/health-care/health-care-unittest.spec.js]: Did you mean `from` instead of `form`?

File: _health-care/_js/_form.js [hunk 3]
```diff
@@ -1335,3 +1346,18 @@ export function build1010ezXml(theForm) {
 
   return xmlDoc;
 }
+
+// Exports that for use in unittests only. Nest them one layer lower so the
```
Comment (akainic) [inline _health-care/_js/_form.js]: I think you're missing the word `are`: `Exports that are for use`

**Comment by akainic** [comment]
@awong-dev Reviewed and everything makes sense to me!

File: spec/javascripts/health-care/health-care-unittest.spec.js (hunk 2 above)
Comment (awong-dev) [inline spec/javascripts/health-care/health-care-unittest.spec.js]: yup.

File: _health-care/_js/_form.js (hunk 3 above)
Comment (awong-dev) [inline _health-care/_js/_form.js]: done.

**Comment by awong-dev** [comment]
PTAL

**Comment by akainic** [comment]
LGTM!

**Pull request merged**